apps/src/music/redux/musicRedux.ts

```diff
@@ -27,6 +28,7 @@ interface MusicState {
   showInstructions: boolean;
   instructionsPosition: InstructionsPosition;
   isBeatPadShowing: boolean;
+  currentProgressState: ProgressState;
```

Comment:
We wanted `ProgressManager` to be lab-independent.  Does it feel right that the state is then in `musicRedux`, as opposed to some lab-independent namespace?

Reply:
Oh great question, that's a good point. I think I see this going in a couple directions...if we do think we'll likely set up some sort of LabLab lgeneric redux reducer, then it would make more sense to store progress state there. Furthermore, if we want musiclab's instructions component to become more generic, as something you can drop into any lab, then any state that drives it (including the progress state) should live in the generic lab-level reducer.

That being said, given some of our recent conversations around keeping redux strictly in views - if we do go down the path where LabLab mostly functions as a library/API (similar to how Maddie's ProgressManager is functioning today), then I can see a situation where all view state management is delegated to the individual labs. In which case, each lab reducer would listen to its own progress state, and use it in its views however it wants. That would allow each lab to have more of a custom treatment of progress and associated views. But I'm not sure if we that's the direction we want to head in.

For now, since we don't have any sort of generic LabLab redux set up just yet, I'm good with 1) creating a separate reducer that we can migrate later on, 2) keeping this as is with a TODO to migrate later on, or 3) deferring this change until later. It's not really that urgent, especially as progress is an area we'll continue to experiment with.

cc @molly-moen for thoughts as well!

Reply:
I'm not aware of our plans relating to progress here, but I think Sanchit's options make sense. If we are pretty confident we can reuse this progress state across labs, then putting it in a generic reducer (even if for now it's in the music lab folder) makes sense to me. If we think it might be music lab specific, leaving it here makes sense.

Reply:
No strong feelings yet, but I want to start integrating the newer progression approach into the level system soon so we can start fleshing out some of these relationships.  In the meantime, any of the above sounds fine; we can easily adjust as we learn more.

Reply:
Cool sounds good. I'll leave a comment for now, and we can revisit this if/when progress moves out of music lab.